app/lib/active_storage/service/druid_disk_service.rb

```diff
@@ -0,0 +1,21 @@
+# typed: false
+# frozen_string_literal: true
+
+require 'active_storage/service/disk_service'
+
+# Use a Druid-based disk service
+module ActiveStorage
+  class Service
+    # Provides a druid-based tree path
+    class DruidDiskService < DiskService
+      private
+
+      def folder_for(key)
+        work = ActiveStorage::Blob.find_by(key: key).attachments.first.record.work
```

Comment:
Should we handle the case where this happens to be called on an unattached blob?

Reply:
Should/n't that be an error condition?